src/crypto/public_key/elliptic/ecdh.js

```diff
@@ -340,10 +353,50 @@ async function ellipticPublicEphemeralKey(curve, Q) {
   return { publicKey, sharedKey };
 }
 
+/**
+ * Generate ECDHE secret from private key and public part of ephemeral key using nodeCrypto
+ *
+ * @param  {Curve}                  curve          Elliptic curve object
+ * @param  {Uint8Array}             V              Public part of ephemeral key
+ * @param  {Uint8Array}             d              Recipient private key
+ *
+ * @returns {Promise<{secretKey: Uint8Array, sharedKey: Uint8Array}>}
+ * @async
+ */
+async function nodePrivateEphemeralKey(curve, V, d)
+{
+  const recipient = nodeCrypto.createECDH(curve.node.node);
+  recipient.setPrivateKey(d);
+  const sharedKey = new Uint8Array(recipient.computeSecret(V));
+  const secretKey = new Uint8Array(recipient.getPrivateKey());
+  return { secretKey, sharedKey };
+}
+
+/**
+ * Generate ECDHE ephemeral key and secret from public key using nodeCrypto
+ *
+ * @param  {Curve}                  curve        Elliptic curve object
+ * @param  {Uint8Array}             Q            Recipient public key
+ *
+ * @returns {Promise<{publicKey Uint8Array, sharedKey Uint8Array}>}
+ * @async
+ */
+async function nodePublicEphemeralKey(curve, Q)
+{
+  const sender = nodeCrypto.createECDH(curve.node.node);
+  sender.generateKeys();
+  //below would throw different errors depending on version of node
```

Comment:
It's probably better to explain this in the test case (something like "computeSecret inside nodePublicEphemeralKey throws different...") instead of here, as it's otherwise not really relevant to this code.